Fitter/src/net/relinc/fitter/GUI/HomeController.java

```diff
@@ -209,6 +209,36 @@ public void handle(MouseEvent event) {
 		
 		
 		
+	}
+
+	public void setFitterFromTrimSelection(Double xAxisBegin, Double xAxisEnd){
+		FitableDataset theData = datasetsListView.getSelectionModel().getSelectedItem();
+		theData.setBeginFromXValue(xAxisBegin);
+		theData.setEndFromXValue(xAxisEnd);
+		renderCharts();
+
+		FitableDataset f = getCurrentDataset();
+		if(f == null)
+			return;
+		int fit = (int)polynomialOrderScrollBar.getValue();
+		f.setPolynomialFit(fit);
+		f.renderFittedData();
+		renderCharts();
+
+		double maxY = -Double.MAX_VALUE;
+		double minY = Double.MAX_VALUE;
+		for(int i = 0; i < theData.origX.size(); i++){
+			if( theData.origX.get(i) >= xAxisBegin && theData.origX.get(i) <= xAxisEnd) {
+				if( theData.origY.get(i) > maxY ) { maxY = theData.origY.get(i); }
+				if( theData.origY.get(i) < minY ) { minY = theData.origY.get(i); }
+			}
+		}
```

Comment:
If you wanted to use some fancy stream code, you could use
```java
		DoubleSummaryStatistics stats = theData.origX.stream().filter(v -> v >= xAxisBegin && v <= xAxisEnd).mapToDouble(n -> n).summaryStatistics();
		double maxY = stats.getMax();
		double minY = stats.getMin();
```
but yours works perfectly fine, no need to change